src/scripts/clientUtils.js

```diff
@@ -1,35 +1,41 @@
+import * as webSettings from 'webSettings';
 
 export function getCurrentUser() {
     return window.ApiClient.getCurrentUser(false);
 }
 
-//TODO: investigate url prefix support for serverAddress function
-export function serverAddress() {
-    if (AppInfo.isNativeApp) {
-        const apiClient = window.ApiClient;
-
-        if (apiClient) {
-            return apiClient.serverAddress();
-        }
+// TODO: investigate url prefix support for serverAddress function
+export async function serverAddress() {
+    const apiClient = window.ApiClient;
 
-        return null;
+    if (apiClient) {
+        return Promise.resolve(apiClient.serverAddress());
     }
 
-    const urlLower = window.location.href.toLowerCase();
-    const index = urlLower.lastIndexOf('/web');
+    let current = await window.connectionManager.getAvailableServers().then(servers => {
+        if (servers.length !== 0) {
+            return Promise.resolve(servers[0].ManualAddress);
+        }
+    });
 
-    if (index != -1) {
-        return urlLower.substring(0, index);
-    }
+    if (current) return Promise.resolve(current);
 
-    const loc = window.location;
-    let address = loc.protocol + '//' + loc.hostname;
+    let urls = [];
+    urls.push(`${window.location.origin}/System/Info/Public`);
+    urls.push(`${window.location.protocol}//${window.location.hostname}:8096/System/Info/Public`);
```

Comment:
hardcoded port? :confused:

Reply:
Not hardcoded so much as a "sane default" to search for common server locations. The alternatives would be not searching at all, or searching every single port on the origin to check for servers.

Reply:
But can it be replaced easily? Sane default out of a config file would be nicer than in code IMO. Would then be scalable to automate the creation of multiple frontends if the remote port was altered.

Reply:
How about putting it in config or controllable via env var on server? Most common use-case scenario is probably for coupling with a self-managed Jellyfin server.

Reply:
You can specify any number of servers in the config file and they will all get queried as well. This is to accomodate the most standard configuration.